lib/usd/hdMaya/adapters/materialAdapter.cpp

```diff
@@ -552,6 +559,52 @@ class HdMayaShadingEngineAdapter : public HdMayaMaterialAdapter {
         }
     }
 
+    HdTextureResource::ID GetTextureResourceID(
+        const TfToken& paramName) override {
+        TF_DEBUG(HDMAYA_ADAPTER_MATERIALS)
+            .Msg("HdMayaShadingEngineAdapter::GetTextureResourceID(%s): %s\n",
+                    paramName.GetText(), GetID().GetText());
+        auto fileObj = GetConnectedFileNode(_surfaceShader, paramName);
+        if (fileObj == MObject::kNullObj) { return HdTextureResource::ID(-1); }
+        return _GetTextureResourceID(
+            fileObj, GetFileTexturePath(MFnDependencyNode(fileObj)));
+    }
+
+
+    HdTextureResourceSharedPtr GetTextureResource(
+        const TfToken& paramName) override {
+        TF_DEBUG(HDMAYA_ADAPTER_MATERIALS)
+            .Msg("HdMayaShadingEngineAdapter::GetTextureResource(%s): %s\n",
+                    paramName.GetText(), GetID().GetText());
+        auto fileObj = GetConnectedFileNode(_surfaceShader, paramName);
+        if (fileObj == MObject::kNullObj) { return {}; }
+        if(GetDelegate()->IsHdSt()) {
```

Comment:
Maybe move this check before we try to get `fileObj` so we can bail out earlier if it fails?

Reply:
Addressed in the dev-branch PR:

https://github.com/Autodesk/maya-usd/pull/231/commits/6549209f9b47035641d63dd4afaedfbf9a84d46b

I'll migrate (or merge) all those changes over to this branch once #231 is merged - assuming we still need refactoring_sandbox at that point!

Reply:
We have one more PR to do - MayaReference schema move. After that, would be good to have ability to confirm that nothing else was forgotten.

So yes, please merge changes back once #231 is merged (I will let you pick the best strategy for it).